### ISSUE TITLE
Fails to build on s390x

### ISSUE THREAD

**Issue by iliastsi** [rblaze/haskell-dbus#26]
haskell-dbus fails to build on Debian's s390x architecture with the following error:

```
[11 of 11] Compiling DBus.TH          ( lib/DBus/TH.hs, dist-ghc/build/DBus/TH.o )

lib/DBus/TH.hs:1:1: error:
    Exception when trying to run compile-time code:
      Prelude.head: empty list
    Code: generateFromFilePath
            defaultGenerationParams
              {genBusName = Just dbusName, genObjectPath = Just dbusPath}
            $ "idlxml" </> "dbus.xml"
  |
1 | {-# LANGUAGE OverloadedStrings #-}
  | ^
```

You can see the build logs [here](https://buildd.debian.org/status/fetch.php?pkg=haskell-dbus&arch=s390x&ver=1.0.1-3&stamp=1540806960&raw=0).

The same has been reported at Fedora's bug [tracker](https://partner-bugzilla5.redhat.com/show_bug.cgi?id=1442648#c15).

Could you please take a look? Let me know if I can provide any further information.

**Comment by rblaze** [comment]
I'll take a look.

**Comment by rblaze** [comment]
Is there any way to test my diff on s390? It's not like TravisCI supports this one.

**Comment by iliastsi** [comment]
Thanks for taking a look :)

Requesting access to Debian's porter machines usually takes time, but you can emulate s390x using [QEMU](https://wiki.qemu.org/Documentation/Platforms/S390X). You can also send me any patches you want to try and I will report back with the results.

**Comment by rblaze** [comment]
QEMU will work just fine, if I'll be able to set it up :)

**Comment by rblaze** [comment]
It looks like qemu in my Ubuntu distribution has a problem Debian fixed couple years ago: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=832897

I'm trying to find a workaround.

**Comment by iliastsi** [comment]
FYI, https://ghc.haskell.org/trac/ghc/ticket/15933

**Comment by rblaze** [comment]
Wow, ghc bug is fun :)

I'll have new version with a fix published by the end of the week, probably earlier.

**Comment by rblaze** [comment]
I just published v1.1.0 to hackage, it should work on s390x and other 32-bit platforms too.

**Comment by iliastsi** [comment]
Thank you, for fixing it so quickly :)
